### PR TITLE
Mpt2test memory state support

### PR DESCRIPTION
**Pull request by ArnauBigas**
This PR adds support for replicating the memory state of an original benchmark in a microbenchmark. To do so, it uses a new, slightly different, assembly wrapper which organizes data and code into sections. These are referenced by an automatically generated linker script, indicating the actual addresses into which this data/code is to be located.

@rbertran I tried to apply the feedback you've given me over time, but I might've forgot some. If that is the case, just let me know and I'll try implementing it next week ;)

### CODE REVIEW

**Comment by rbertran** [comment]
@ArnauBigas I did some minor updates to merged the branch with the latest master branch and fix the CI setup.

Then, I realized that the functionality implemented in `mp_mpt2test` are very similar to the one implemented in the `mp_mpt2elf` . The first has been tested on your side on a RISCV platform, whereas the second has been tested on POWER and Z platforms. 

The solution to avoid replicating similar code would be to cleanup/refactor/merge the functionality so that the same tool is valid across the three platforms but before doing further changes, I wanted to check with you to avoid "breaking" your current RISCV setup.

Ideally, I'd want to proceed as following: 1st, I'll integrate the your contributions to the `mp_mpt2elf`, 2nd: we validate that it works for all the tests (i.e. not new regressions on your tests or my tests) and finally, I'd remove the `mp_mpt2test` one. 

Let me know, and I'll work accordingly.

File: src/microprobe/target/env/__init__.py [hunk 1]
```diff
@@ -215,6 +215,11 @@ def default_wrapper(self):
         """ """
         raise NotImplementedError
 
+    @abc.abstractproperty
+    def default_endless_wrapper(self):
```
Comment (rbertran) [inline src/microprobe/target/env/__init__.py:219]: I'd not add a new property at the API --> implies too many changes in other several environment backends

**Comment by ArnauBigas** [comment]
Sorry, I had completely missed your last message. I see what you mean by the two tools being similar, please go ahead with the merge, I will do whatever testing is necessary on my side.

Speaking of which, I have finally ported all the tools necessary on our side to begin testing this code on Power. Since you open sourced the power backend for microprobe (much appreciated!), I can proceed with the validation. I don't expect to need to change the code in this PR. If I do have to change something, I will try to keep the changes in a separate branch (and tell you beforehand) so that I don't interfere with your work.

**Comment by rbertran** [comment]
@ArnauBigas , I've integrated the RISCV functionality of `mp_mpt2test` into` mp_mpt2elf` tool. The following command: 

`mp_mpt2elf.py -T riscv_v22-riscv_generic-riscv64_linux_gcc -t <input_mpt_file>  -O <output.s> --raw-bin --reset --compiler <your_compiler>`

should create ELF binaries that can be run in linux out of the mpt files. One change to take into account is that I've added the Program Counter (PC) register (which is user-visible) to the RISCV target definition. As a result, the PC register value should be also part of the Microprobe state file accompanying the mpt file. For consistency, this will require that other tools (e.g. Chopstix) to also dump that register in the mpt/mps files generated (not a big deal I think, since that value is already provided in the default_code_address). 

Please give a try at this branch whenever you have some time to make sure your still maintain the functionality you have before with the `mpt2test` tool. Once you confirm, I'll remove that tool, and remove some other unused classes.

**Comment by ArnauBigas** [comment]
That looks great! Right now I'm a bit busy, but hopefully I'll be able to test this by next week.
Thanks!

**Pull request merged**